### ISSUE TITLE
External document refs should have a SHA1 instead of a SHA256

### ISSUE THREAD

**Issue by goneall** [swinslow/cmake-spdx#8]
It is admittedly a bit hard to follow in the spec, but my reading is the external document references must use SHA1.

This is based on [section 2.6](https://spdx.github.io/spdx-spec/2-document-creation-information/#26-external-document-references) subsection 2.6.4:

```
[Checksum] is a checksum of the external document following the checksum

format defined in section 4.4.
```

In [section 4.4](https://spdx.github.io/spdx-spec/4-file-information/#44-file-checksum) subsection 4.4.3:

```
4.4.3 Cardinality: Mandatory, one SHA1, others may be optionally provided.
```

**Comment by goneall** [comment]
There is also a missing space between the algorithm and algorithm value.  Again, this is a bit hard to read in the spec, but the [Java tools](https://github.com/spdx/tools-java) use the space as a delimiter between the tag/value fields.  If we decide the space isn't needed, we should raise an issue for the Java tools.

**Comment by swinslow** [comment]
Thanks for this @goneall. From our separate conversation on this, understood about changing this from SHA256 to SHA1.

On the spacing comment, apologies for nitpicking :) but I'm not sure I gathered this from just reading the spec. [Section 4.4.5](https://spdx.github.io/spdx-spec/4-file-information/#44-file-checksum) describes the separator as being `":"`, not `": "`. I guess I had assumed that the separator was the colon, and any extraneous whitespace would be optional and ignorable (I may have made similar assumptions throughout the Golang tools tag-value parser, though the tag-value writer appears to add a single space as you're describing).

I do see though that the examples show one space between the colon and the actual hash value. Is it assumed that it should be a single whitespace? If so, I'll update it here and we may want to update the text in the spec to make that explicit as well.

**Comment by goneall** [comment]
@swinslow Good point on the spec - I agree that there should not be a space required.  I'll open an issue for the Java tools that does the verification and creates the examples in the spec repo.

**Comment by fishface60** [comment]
I came across this issue while parsing some SPDX documents using https://github.com/spdx/tools-python

The specification isn't explicit about which hash to use since it just forwards on to the package hash which can have multiple, but https://github.com/spdx/tools-python/blob/b7f9a3defe8b0af1414cd75f1447ec242428f1d0/src/spdx_tools/spdx/parser/tagvalue/parser.py#L288 is where it's implemented to expect SHA1 only.
